drivers/hdlc_rcp_if/hdlc_rcp_if_nxp.c

```diff
@@ -0,0 +1,144 @@
+/*                                  Definitions                               */
+/* -------------------------------------------------------------------------- */
+
+#define DT_DRV_COMPAT nxp_hdlc_rcp_if
+
+#define LOG_MODULE_NAME hdlc_rcp_if_nxp
+#define LOG_LEVEL       CONFIG_HDLC_RCP_IF_DRIVER_LOG_LEVEL
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#define HDLC_RCP_IF_IRQ_N        DT_INST_IRQ_BY_NAME(0, hdlc_rcp_if_int, irq)
+#define HDLC_RCP_IF_IRQ_P        DT_INST_IRQ_BY_NAME(0, hdlc_rcp_if_int, priority)
+#define HDLC_RCP_IF_WAKEUP_IRQ_N DT_INST_IRQ_BY_NAME(0, wakeup_int, irq)
+#define HDLC_RCP_IF_WAKEUP_IRQ_P DT_INST_IRQ_BY_NAME(0, wakeup_int, priority)
+
+struct ot_hdlc_rcp_context {
+	struct net_if *iface;
+	struct openthread_context *ot_context;
+};
+
+/* -------------------------------------------------------------------------- */
+/*                             Private prototypes                             */
+/* -------------------------------------------------------------------------- */
+extern int32_t hdlc_rcp_if_handler(void);
+extern int32_t hdlc_rcp_if_wakeup_done_handler(void);
+
+/* -------------------------------------------------------------------------- */
+/*                             Private functions                              */
+/* -------------------------------------------------------------------------- */
+
+static void hdlc_iface_init(struct net_if *iface)
+{
+	struct ot_hdlc_rcp_context *ctx = net_if_get_device(iface)->data;
+	otExtAddress eui64;
+
+	/* HDLC RCP interface Interrupt */
+	IRQ_CONNECT(HDLC_RCP_IF_IRQ_N, HDLC_RCP_IF_IRQ_P, hdlc_rcp_if_handler, 0, 0);
+	irq_enable(HDLC_RCP_IF_IRQ_N);
+
+	/* Wake up done interrupt */
+	IRQ_CONNECT(HDLC_RCP_IF_WAKEUP_IRQ_N, HDLC_RCP_IF_WAKEUP_IRQ_P,
+		    hdlc_rcp_if_wakeup_done_handler, 0, 0);
+	irq_enable(HDLC_RCP_IF_WAKEUP_IRQ_N);
+
+	ctx->iface = iface;
+
+	ieee802154_init(iface);
+
+	ctx->ot_context = net_if_l2_data(iface);
+
+	otPlatRadioGetIeeeEui64(ctx->ot_context->instance, eui64.m8);
+	net_if_set_link_addr(iface, eui64.m8, OT_EXT_ADDRESS_SIZE, NET_LINK_IEEE802154);
+}
+
+static int hdlc_register_rx_cb(hdlc_rx_callback_t hdlc_rx_callback, void *param)
+{
+	int ret = 0;
+	int status = 0;
+
+	do {
+		status = PLATFORM_InitHdlcInterface(hdlc_rx_callback, param);
+		if (status < 0) {
+			LOG_ERR("HDLC RX callback registration failed");
+			ret = EIO;
+			break;
+		}
+
+	} while (false);
+
+	return ret;
+}
+
+static int hdlc_send(const uint8_t *frame, uint16_t length)
+{
+	int ret = 0;
+	int status = 0;
+
+	status = PLATFORM_SendHdlcMessage((uint8_t *)frame, length);
+	if (status < 0) {
+		LOG_ERR("HDLC send frame failed");
+		ret = EIO;
+	}
+
+	return ret;
+}
+
+static int hdlc_deinit(void)
+{
+	int ret = 0;
+	int status = 0;
+
+	status = PLATFORM_TerminateHdlcInterface();
+	if (status < 0) {
+		LOG_ERR("Failed to shutdown OpenThread controller");
+		ret = EIO;
+	}
+
+	return ret;
+}
+
+static struct hdlc_api hdlc_api = {.iface_api.init = hdlc_iface_init,
```

Comment:
Do not use a name that is the same as the struct type, also this should be `const`.

I wonder why compliance doesn't complain here?

Reply:
Agree & resolved